include/mxnet/base.h

```diff
@@ -307,6 +312,21 @@ inline Context Context::GPU(int32_t dev_id) {
   return Create(kGPU, dev_id);
 }
 
+inline int32_t Context::GetGPUCount() {
+#if MXNET_USE_CUDA
+  int32_t count;
+  cudaError_t e = cudaGetDeviceCount(&count);
+  if (e == cudaErrorNoDevice) {
+    return 0;
+  }
+  CHECK_EQ(e, cudaSuccess) << " CUDA: " << cudaGetErrorString(e);
+  return count;
+#else
+  LOG(FATAL) << "Please compile with CUDA support to query the number of GPUs.";
```

Comment:
I agree with @marcoabreu 
it's probably better to just return 0 when mxnet is not compiled with gpu.

Reply:
that's what I had originally before you suggested I should raise an error. Quoting:
```
piiswrong reviewed 22 days ago
Machine has no gpu and mxnet not compiled with gpu should be treated separately. This should raise an error too.
```
I will happily change this back though.

Reply:
no worries. will change this back.